lib/util.js

```diff
@@ -684,14 +684,32 @@ function formatValue(ctx, value, recurseTimes) {
   }
 
   ctx.seen.push(value);
-  const output = formatter(ctx, value, recurseTimes, keys);
-
+  let output;
+  // This corresponds to a depth of at least 333 and likely 500.
+  if (ctx.indentationLvl < 1000) {
+    output = formatter(ctx, value, recurseTimes, keys);
+  } else {
+    try {
+      output = formatter(ctx, value, recurseTimes, keys);
+    } catch (err) {
+      if (errors.isStackOverflowError(err)) {
+        ctx.seen.pop();
+        process.emitWarning(
+          'Inspection reached the maximum call stack size. Incomplete ' +
+          'inspected object returned.'
+        );
```

Comment:
I don’t think we should be printing this warning, it’s not a good idea to have side effects (especially unexpected) like stdio coming from `util.inspect()`.

Reply:
Without the warning though, it would completely be silent and users would very likely not realize this at all. I guess it would not really be important / bad in this case but I believe it would be good to provide some kind of notification about it and I can not think of any alternative right now.

Reply:
@BridgeAR Can we add this to the output instead? That seems to make more sense to me, e.g. changing in the line below from `[${constructor || tag || 'Object'}]` to `[${constructor || tag || 'Object'}: Maximum call stack size exceeded]`?

Reply:
I think that is a viable solution. Should we maybe even be more verbose by writing e.g., `[...: Inspection interrupted prematurely. Maximum call stack size exceeded.]`?

I also think about adding the actually reached depth. Opinions about that?